components/measure-upload/MeasureRepositoryUpload.tsx

```diff
@@ -0,0 +1,235 @@
+      message,
+      timestamp: new Date().toISOString(),
+      attemptedBundleDisplay,
+      isValueSetMissingError
+    });
+    showNotification({
+      icon: <IconAlertCircle />,
+      title: 'Bundle upload failed',
+      message: 'There was an issue uploading your bundle. Please correct the issues listed below.',
+      color: 'red'
+    });
+    setIsLoadingPackage(PACKAGE_STATES.FAIL);
+  };
+
+  /*
+    Handles updating of the Measure Repository Service url field and associated state variables
+  */
+  function handleMrsUrlChange(event: React.ChangeEvent<HTMLInputElement>) {
+    if (measureSelectIsOpen) {
+      setMeasureSelectIsOpen(false);
+    }
+    setMeasureBundle(mb => ({
+      ...mb,
+      measureRepositoryUrl: event.currentTarget.value,
+      selectedMeasureId: null,
+      displayMap: {},
+      ...(!mb.isFile ? { content: null } : undefined)
+    }));
+    setIsLoadingPackage(PACKAGE_STATES.NONE);
+  }
+
+  /*
+    Takes the entries from a Measure retrieval response and creates a map from Measure id
+    to Measure display label for Measure select component
+  */
+  function createMeasureDisplayMap(
+    displayMap: Record<string, string>,
+    { resource: measure }: { resource: fhir4.Measure }
+  ) {
+    let label: string;
+    const officialIdentifier = measure.identifier?.find(
+      identifier =>
+        identifier.use === 'official' && identifier.system === 'http://hl7.org/fhir/cqi/ecqm/Measure/Identifier/cms'
+    );
+    if (officialIdentifier?.value) {
+      label = measure.version ? `${officialIdentifier.value}|${measure.version}` : officialIdentifier.value;
+    } else if (measure.name) {
+      label = measure.version ? `${measure.name}|${measure.version}` : measure.name;
+    } else {
+      label = measure.id as string;
+    }
+    displayMap[measure.id as string] = label;
+    return displayMap;
+  }
+
+  /*
+    Sends a `GET` request to the `/Measure` endpoint of the specified Measure Repository Service.
+    Parses and stores the measureIds if successful
+  */
+  async function retrieveMeasures() {
+    setIsLoadingIds(true);
+    try {
+      const response = await fetch(`${measureRepositoryUrl}/Measure`);
+      const responseBody = await response.json();
+      if (response.status === 200) {
+        const displayMap = responseBody.entry.reduce(createMeasureDisplayMap, {});
+        setMeasureBundle(mb => ({ ...mb, displayMap }));
+        if (!measureSelectIsOpen) {
+          setMeasureSelectIsOpen(true);
+        }
+      } else {
+        showNotification({
+          icon: <IconAlertCircle />,
+          title: 'Failed reaching out to server',
+          message: `Server responded with code: ${response.status}:${response.statusText}`,
+          color: 'red'
+        });
+      }
+    } catch (e) {
+      showNotification({
+        icon: <IconAlertCircle />,
+        title: 'Failed reaching out to server',
+        message: `Could not connect to server with url: ${measureRepositoryUrl}`,
```

Comment:
As we have it right now, we can click on "Get Measures" without providing any url input, and this error will be thrown and say `"Could not connect to server with url: "` since `measureRepositoryUrl` is an empty string. Instead of this, should we add functionality that will instead gray out the "Get Measures" button when the search bar is empty? That way we will never have this error thrown with an empty `measureRepositoryUrl`

Could also add this as a follow-up task to the backlog if it would cause too much overhead... still getting back into the frontend mindset so not sure how difficult this would be to implement, given the current functionality.